cpp/louvain_module/algorithm/louvain.cpp

```diff
@@ -0,0 +1,149 @@
+#include "louvain.hpp"
+
+namespace {
+
+constexpr int kReplaceMap = 0;
+constexpr int kThreadsOpt = 1;
+constexpr int kNumColors = 16;
+
+std::vector<std::uint64_t> GrappoloCommunityDetection(graph *grappolo_graph, bool coloring,
+                                                      std::uint64_t min_graph_size, double threshold,
+                                                      double coloring_threshold) {
+  auto number_of_vertices = grappolo_graph->numVertices;
+
+  int num_threads = 1;  // Default is one thread
+#pragma omp parallel
+  { num_threads = omp_get_num_threads(); }
+
+  // Initialize clustering array
+  auto *cluster_array = (long *)malloc(number_of_vertices * sizeof(long));
+#pragma omp parallel for
+  for (long i = 0; i < number_of_vertices; i++) {
+    cluster_array[i] = -1;
+  }
+
+  if (coloring) {
+    runMultiPhaseColoring(grappolo_graph, cluster_array, coloring, kNumColors, kReplaceMap, min_graph_size, threshold,
+                          coloring_threshold, num_threads, kThreadsOpt);
+  } else {
+    runMultiPhaseBasic(grappolo_graph, cluster_array, kReplaceMap, min_graph_size, threshold, coloring_threshold,
+                       num_threads, kThreadsOpt);
+  }
+
+  // Store clustering information in vector
+  std::vector<std::uint64_t> result;
+  for (long i = 0; i < number_of_vertices; ++i) {
+    result.emplace_back(cluster_array[i]);
+  }
+  // Detach memory, no need to free graph instance, it will be removed inside algorithm
+  free(cluster_array);
+
+  // Return empty vector if algorithm has failed
+  return result;
+}
+
+void LoadUndirectedEdges(const mg_graph::GraphView<> &memgraph_graph, graph *grappolo_graph) {
+  int num_threads = 1;
+#pragma omp parallel
```

Comment:
Unused var
```suggestion
```

Reply:
This variable is here to set the number of threads in the parallel implementation as in documentation written here: https://www.ibm.com/docs/en/zos/2.3.0?topic=processing-pragma-omp-parallel

Reply:
But according to the docs (https://www.ibm.com/docs/en/xl-c-aix/13.1.0?topic=processing-pragma-omp-parallel) the num_threads is a omp clause and it should fall under the #pragma macro, like this:
```suggestion
#pragma omp parallel num_threads(1)
```
Some examples (https://nanxiao.me/en/the-pitfalls-of-using-openmp-parallel-for-loops/ and https://github.com/MicrosoftDocs/cpp-docs/blob/master/docs/parallel/openmp/a-examples.md#a14-a-flush-directive-without-a-list)

Reply:
After reading a bit about OpenMP, it seems that it is using the dynamic variable for setting the number of threads, therefore there is no need to set it explicitly, otherwise, we might encounter an error of resource not being available at the time of creation.